src/vs/workbench/parts/files/browser/views/explorerViewer.ts

```diff
@@ -111,6 +161,55 @@ export class FileDataSource implements IDataSource {
 		}
 	}
 
+	private resolveVirtualDirectories(files: FileStat[]) {
+		if (!this.enableVirtualDirectories) {
+			return;
+		}
+
+		files = files.filter(x => !x.isDirectory);
```

Comment:
Effectively, you're asking `what are my children`, whereas I am asking `what is my parent`. My logic always takes the first answer (`[0]`) to that question and thus avoids any duplicate results. I was considering adding a priority system for the rules, but as it is it works first-come, first-serve which is fine for now.